input/drivers/udev_input.c

```diff
@@ -505,38 +505,39 @@ static bool udev_input_add_device(udev_input_t *udev,
 
    strlcpy(device->devnode, devnode, sizeof(device->devnode));
 
-   /* Touchpads report in absolute coords. */
-   if (type == UDEV_INPUT_TOUCHPAD)
-   {
-      if (ioctl(fd, EVIOCGABS(ABS_X), &absinfo) < 0 ||
-            absinfo.minimum >= absinfo.maximum)
-         goto error;
-
-      device->mouse.x_min = absinfo.minimum;
-      device->mouse.x_max = absinfo.maximum;
-
-      if (ioctl(fd, EVIOCGABS(ABS_Y), &absinfo) < 0 ||
-          absinfo.minimum >= absinfo.maximum)
-         goto error;
-
-      device->mouse.y_min = absinfo.minimum;
-      device->mouse.y_max = absinfo.maximum;
-   }
+  
    /* UDEV_INPUT_MOUSE may report in absolute coords too */
-   else if (type == UDEV_INPUT_MOUSE && ioctl(fd, EVIOCGABS(ABS_X), &absinfo) >= 0)
+   if (type == UDEV_INPUT_MOUSE || type == UDEV_INPUT_TOUCHPAD )
    {
-      if (absinfo.minimum >= absinfo.maximum)
-         goto error;
-
-      device->mouse.x_min = absinfo.minimum;
-      device->mouse.x_max = absinfo.maximum;
-
-      if (ioctl(fd, EVIOCGABS(ABS_Y), &absinfo) < 0 ||
-          absinfo.minimum >= absinfo.maximum)
-         goto error;
+      if (ioctl(fd, EVIOCGABS(ABS_X), &absinfo) >= 0)
+      {      
+	
+        if (absinfo.minimum >= absinfo.maximum )
+      	{
+          device->mouse.x_min = 0;
```

Comment:
Should be -1 to 1 to avoid potential divide by zero scenario.

Reply:
which one do you wnt zero and which one do you want -1